### PR TITLE
NB implementation of the admin ConfigurationService

### PR DESCRIPTION
**Pull request by tomikazi**

### CODE REVIEW

File: pkg/manager/manager.go [hunk 1]
```diff
@@ -133,7 +136,7 @@ func (m *Manager) startNorthboundServer(
 
 	s.AddService(logging.Service{})
 
-	adminService := admin.NewService(networkChangesStore, networkSnapshotStore, deviceSnapshotStore, pluginRegistry)
+	adminService := admin.NewService(transactionsStore, configurationsStore, pluginRegistry, networkChangesStore, networkSnapshotStore, deviceSnapshotStore)
```
Comment (adibrastegarnia) [inline pkg/manager/manager.go:139]: I think we should remove networkChangeStore, snaphort store, deviceStore from the arguments and clean up admin service.

File: pkg/manager/manager.go [hunk 2]
```diff
@@ -278,6 +281,9 @@ func (m *Manager) Start() error {
 		return err
 	}
 
+	// TODO: placeholder for creation of the transactions store
+	var transactions transaction.Store
```
Comment (adibrastegarnia) [inline pkg/manager/manager.go:285]: Go ahead and create a store like Configuration store and start it. You don't need to put a place holder.

File: pkg/northbound/admin/configuration.go [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+		if err != nil {
+			log.Warnf("ListConfigurations %+v failed: %v", req, err)
+			return errors.Status(err).Err()
+		}
+	}
+	return nil
+}
+
+// WatchConfigurations provides stream with events representing configuration changes
+func (s Server) WatchConfigurations(req *admin.WatchConfigurationsRequest, stream admin.ConfigurationService_WatchConfigurationsServer) error {
```
Comment (adibrastegarnia) [inline pkg/northbound/admin/configuration.go:55]: Why some of the server instances are pointer some of them are not?

**Review by adibrastegarnia — CHANGES_REQUESTED** [review]

File: pkg/manager/manager.go (hunk 2 above)
Comment (adibrastegarnia) [reply pkg/manager/manager.go:285]: I will rebase

File: pkg/northbound/admin/configuration.go (hunk 3 above)
Comment (tomikazi) [reply pkg/northbound/admin/configuration.go:55]: Fixed.

File: pkg/manager/manager.go (hunk 1 above)
Comment (tomikazi) [reply pkg/manager/manager.go]: Cleaned up as much as possible at this point in time. Left notes in other places.

File: pkg/manager/manager.go (hunk 2 above)
Comment (tomikazi) [reply pkg/manager/manager.go]: Done.

File: pkg/northbound/admin/transaction.go [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+		watchOpts = append(watchOpts, transaction.WithReplay())
+	}
+
+	ch := make(chan configapi.TransactionEvent)
+	if err := s.transactionsStore.Watch(stream.Context(), ch, watchOpts...); err != nil {
+		log.Warnf("WatchTransactionsRequest %+v failed: %v", req, err)
+		return errors.Status(err).Err()
+	}
+
+	return s.streamTransactions(stream, ch)
```
Comment (adibrastegarnia) [inline pkg/northbound/admin/transaction.go:68]: I think we should return the error here and convert it to status before returning it back.

File: pkg/northbound/admin/configuration.go [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+		watchOpts = append(watchOpts, configuration.WithReplay())
+	}
+
+	ch := make(chan configapi.ConfigurationEvent)
+	if err := s.configurationsStore.Watch(stream.Context(), ch, watchOpts...); err != nil {
+		log.Warnf("WatchConfigurationsRequest %+v failed: %v", req, err)
+		return errors.Status(err).Err()
+	}
+
+	return s.streamConfigurations(stream, ch)
```
Comment (adibrastegarnia) [inline pkg/northbound/admin/configuration.go:68]: I think we should return the error here and convert it to status before returning it back.

**Review by adibrastegarnia — CHANGES_REQUESTED** [review]

**Review by adibrastegarnia — APPROVED** [review]

File: pkg/northbound/admin/transaction.go (hunk 4 above)
Comment (tomikazi) [reply pkg/northbound/admin/transaction.go:68]: Fixed

File: pkg/northbound/admin/configuration.go (hunk 5 above)
Comment (tomikazi) [reply pkg/northbound/admin/configuration.go:68]: Fixed

**Pull request merged**